nix/craneCommon.nix

```diff
@@ -72,6 +72,7 @@ craneLib.overrideScope' (self: prev: {
     PROTOC = "${pkgs.protobuf}/bin/protoc";
     PROTOC_INCLUDE = "${pkgs.protobuf}/include";
     CARGO_PROFILE = self.commonProfile;
+    FEDIMINT_BUILD_ALLOW_GIT_FAIL = "true";
```

Comment:
I think we will need to refactor it and move it to a new nix variable, because this one gets sourced into our dev shell:

```
> nix develop
> echo $FEDIMINT_BUILD_ALLOW_GIT_FAIL 
true
```

which is probably not what we want.

But let's land and we can fix later.

Reply:
It still fails in CI. I think we need to install git in the CI environment to get it to pass.